src/winstanley/references/WdlCallBlockDeclarationReference.scala

```diff
@@ -0,0 +1,42 @@
+package winstanley.references
+
+import javax.annotation.Nullable
+
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.{PsiElement, PsiReferenceBase}
+import winstanley.psi.WdlCallableLookup
+import winstanley.structure.WdlImplicits._
+
+final case class WdlCallableDeclarationReference(value: WdlCallableLookup) extends PsiReferenceBase[PsiElement](value, value.getTextRange) {
+
+  /**
+    * Returns the element which is the target of the reference !!! OR NULL IF NOT FOUND !!!
+    *
+    * @return the target element, or null if it was not possible to resolve the reference to a valid target.
+    * @see PsiPolyVariantReference#multiResolve(boolean)
+    */
+  @Nullable
+  override def resolve(): PsiElement = {
+    value.findTaskDeclarationsAvailableInScope.find(d => value.getCalledIdentifier == d.getNameIdentifier.getText).map(_.getNameIdentifier).orNull
+  }
+
+  /**
+    * Returns the array of String, PsiElement and/or LookupElement
+    * instances representing all identifiers that are visible at the location of the reference. The contents
+    * of the returned array is used to build the lookup list for basic code completion. (The list
+    * of visible identifiers may not be filtered by the completion prefix string - the
+    * filtering is performed later by IDEA core.)
+    *
+    * @return the array of available identifiers.
+    */
+  override def getVariants: Array[AnyRef] = Array.empty[AnyRef]
+
+  /**
+    * This override is required to make reference-lookup work.
+    *
+    * It needs a relative range within the PsiElement 'value' to count as the reference,
+    * which in this case is the entire 'WdlVariableLookup' element.
+    */
+  override def getRangeInElement: TextRange = new TextRange(0, value.getTextLength - 1)
```

Comment:
Remember when you noticed that the last character wasn't included in the clickable range? I'd try experimenting with removing this `-1`